src/test/java/com/hotels/bdp/circustrain/bigquery/client/BigQueryMetastoreClientTest.java

```diff
@@ -111,13 +126,28 @@ public void tableExistsReturnsFalseWhenTableDoesntExist() throws TException {
     assertFalse(bigQueryMetastoreClient.tableExists("database", "table"));
   }
 
+  @Test
+  public void getTableWithCachedTable() throws TException, IOException {
+    HiveTableCache cache = new HiveTableCache();
+    setUpSchema();
+    bigQueryMetastoreClient = new BigQueryMetastoreClient(bigQueryMetastore, extractionService, cache, factory);
+
+    when(bigQueryMetastore.getTable(databaseName, tableName)).thenReturn(table);
+    when(factory.newInstance(any(org.apache.hadoop.hive.metastore.api.Table.class)))
+        .thenReturn(unpartitionedTableService);
+    when(unpartitionedTableService.getTable()).thenReturn(hiveTable);
+    org.apache.hadoop.hive.metastore.api.Table result = bigQueryMetastoreClient.getTable(databaseName, tableName);
+    assertThat(cache.contains(databaseName, tableName), is(true));
+    assertThat(cache.get(databaseName, tableName), is(result));
+    result = bigQueryMetastoreClient.getTable(databaseName, tableName);
```

Comment:
Can you assert something on the result? Otherwise it's not clear why it's being called again (unless to prove that an exception isn't thrown which I don't think is the case and wouldn't require an assignment).

Reply:
Thoughts on this?